superset/databases/commands/create.py

```diff
@@ -69,8 +64,9 @@ def run(self) -> Model:
             security_manager.add_permission_view_menu("database_access", database.perm)
             db.session.commit()
         except DAOCreateFailedError as ex:
-            logger.exception(ex.exception)
-            raise DatabaseCreateFailedError()
+            with event_logger.log_context(action=f"db_creation_failed.{ex.exception}"):
```

Comment:
We might want to just use `log` in places like this. The main difference between `log` and `log_context` is that the former, in addition to being a context manager, logs the duration automatically, together with some dashboard/chart metadata that is irrelevant here.

Reply:
true... it also logs the user_id and referrer, so maybe it's still useful?

Reply:
We may want to either 1) refactor `log` to do some of the magic in `log_context` maybe add a bool flag `enrich_with_request_context=False`, or 2) can we just call the context manager with the `with` block? Would that just work? or 3) refactor `log_context` to NOT be be context manager and create a new one say `log_context_manager` (or better name) that's used only when we want duration / with block

Reply:
One problem is that adding the flag `enrich_with_request_context` would be a breaking change (our event logger, for example, would have to be updated).

I think the best way here is:

1. Create a new method in `AbstractEventLogger` called `log_with_context`, that does all the job that `log_context` does but is a function, not a context manager. The method calls `log` at the end, with all the extracted context.
2. Change `AbstractEventLogger.log_context` to use `log_with_context` for the enrichment, removing the shared code between them.
3. In this PR, call the new `log_with_context`. Since it delegates the actual logging to `log` it will work with all existing event loggers, and is not a breaking change..

Eventually I think we should have the event logger itself be a context manager by implementing `__enter__` and `__exit__` methods, so we could call:

```python
with event_logger(action="foo"):
   do_something()
```

Since we can do this by adding methods to `AbstractEventLogger` it would also not be a breaking change.

Reply:
Quick clarification that `enrich_with_request_context ` would be false by default to protect backwards compatibity. But maybe function composition is better than a parameter here, I'm ok with with either pattern. 

There's a need to really clarify/improve the method in `AbstractEventLogger` with good/clear names here between functions we can call, content manager(s) and decorator(s)

Reply:
since we're merging this branch into @hughhhh's feature branch rather than master, @betodealmeida said that he could do the log_context refactor in a separate PR.

Reply:
I'm working on the refactor right now.

@mistercrunch you're right, I forgot that `log` takes a `**kwargs` that would swallow the new argument.